### PR TITLE
Add GitPushType to GitRefMetadata

### PR DESCRIPTION
**Pull request by patrickxb**

### CODE REVIEW

File: go/protocol/keybase1/git.go [hunk 1]
```diff
@@ -133,10 +133,40 @@ func (o GitCommit) DeepCopy() GitCommit {
 	}
 }
 
+type GitPushType int
+
+const (
+	GitPushType_DEFAULT      GitPushType = 0
+	GitPushType_CREATEREPO   GitPushType = 1
+	GitPushType_DELETEBRANCH GitPushType = 2
```
Comment (jzila) [inline go/protocol/keybase1/git.go]: Need one more for a repo rename.

**Comment by jzila** [comment]
Do you want a before/after name parameter for repo renames?

**Comment by patrickxb** [comment]
I haven't seen the designs so will defer to @cecileboucheron on this.

**Comment by jzila** [comment]
Let's add them to the RPC; then we can decide to use them whenever.

File: go/protocol/keybase1/git.go [hunk 2]
```diff
@@ -133,10 +133,43 @@ func (o GitCommit) DeepCopy() GitCommit {
...
+		return v
+	}
+	return ""
+}
+
 type GitRefMetadata struct {
 	RefName              string      `codec:"refName" json:"refName"`
 	Commits              []GitCommit `codec:"commits" json:"commits"`
 	MoreCommitsAvailable bool        `codec:"moreCommitsAvailable" json:"moreCommitsAvailable"`
+	PushType             GitPushType `codec:"pushType" json:"pushType"`
```
Comment (jzila) [inline go/protocol/keybase1/git.go]: As I said in chat, this should actually be in `GitLocalMetadata`.

**Pull request merged**